### PR TITLE
#160703277 Add login route

### PR DESCRIPTION
**Pull request by daniellamarr**
#### What does this PR do?
- This PR allows a user to login on the app
#### Description of Task to be completed?
- When a **POST** request is sent to **localhost:6060/api/v1/auth/login**, a user should login to the platform
- A JSON Web Token should be generated when a user logs in
- A successful request should return a status of 200
#### How should this be manually tested?
- Pull this branch **ft-user-able-sign-into-app-160703277** off this repository
- On your command line, run npm install to install all dependencies for this app
- On your command line, run npm start to start the app
- App would run on port **6060**
- Access the endpoint with a **POST** request on **localhost:6060/api/auth/login**
- REQUEST BODY
  - email -> string `Your Email`
  - password -> string `Password`
#### Any background context you want to provide?
- This app is a **node app** and would require **nodejs** and **npm** installation on your local machine
#### What are the relevant pivotal tracker stories?
[160703277](https://www.pivotaltracker.com/story/show/160703277)
#### Screenshots (if appropriate)
Request Body
Request Body

Response Object
Response Object

#### Questions:
None

### CODE REVIEW

File: server/helpers/checkUser.js [hunk 1]
```diff
@@ -16,7 +19,41 @@ class CheckUser {
...
+    }
+    
+    static loginCredentials (req,resp,next) {
+        db.query(
+			UserQuery.checkUserQuery(req.body.email),
+			(err,res) => {
+                if (res.rows.length < 1) {
+                    return resp.status(401).send({
+                        status:"error",
+                        message:"You have entered a wrong credential"
```
Comment (fob413) [inline server/helpers/checkUser.js]: Please use a better error message for this

File: server/helpers/checkUser.js [hunk 2]
```diff
@@ -16,7 +19,41 @@ class CheckUser {
...
+                }else{
+                    const [user] = res.rows;
+                    const pass = bcrypt.compareSync(
+                        req.body.password,
+                        user.password
+                    );
+                    if (!pass) {
+                        return resp.status(401).send({
+                            status:"error",
+                            message:"You have entered a wrong credential"
```
Comment (fob413) [inline server/helpers/checkUser.js]: Please use a better error message for this

File: server/helpers/checkUser.js (hunk 1 above)
Comment (daniellamarr) [reply server/helpers/checkUser.js]: Alright

File: server/controllers/userControl.js [hunk 3]
```diff
@@ -26,6 +26,14 @@ class UserControl {
             }
         )
     }
+    
+    static userLogin (req,resp) {
+        resp.status(200).send({
+            status: "success",
+            message: `Your are logged in as ${req.email}`,
```
Comment (fob413) [inline server/controllers/userControl.js]: Please change this message to something that does not include the `req.email`.

Comment (daniellamarr) [reply server/controllers/userControl.js]: Okay

**Review by fob413 — APPROVED** [review]

**Pull request merged**